_posts/2018-01-25-totw-36.md

```diff
@@ -0,0 +1,72 @@
+---
+title: "Tip of the Week #36: New Join API"
+layout: tips
+sidenav: side-nav-tips.html
+published: false
+permalink: tips/36
+type: markdown
+order: "036"
+---
+
+Originally published as totw/36 on 2013-03-21
+
+*By Greg Miller [(jgm@google.com)](mailto:jgm@google.com)*
+
+Updated 2018-01-24
+
+*"I got a good mind to join a club and beat you over the head with it." --
+Groucho Marx*
+
+Many of you asked for a new joining API and we heard you. We now have one
+joining function to replace them all, and it is spelled `absl::StrJoin()`.
+You simply give it a collection of objects to be joined and a separator
+ string, and it does the rest. It will work with collections of `string`,
+`absl::string_view`, `int`, `double` -- any type that `absl::StrCat()`
+supports. If you need to join a type that will not `StrCat()`, you can
+also provide a custom `Formatter` for that type; we'll see below how the
+use of a `Formatter` will let us nicely join a map.
+
+Now for some quick examples:
+
+```c++
+std::vector<string> v = {"a", "b", "c"};
+string s = absl::StrJoin(v, "-");
+// s == "a-b-c"
+
+std::vector<absl::string_view> v = {"a", "b", "c"};
+string s = absl::StrJoin(v.begin(), v.end(), "-");
+// s == "a-b-c"
+
+std::vector<int> v = {1, 2, 3};
+string s = absl::StrJoin(v, "-");
+// s == "1-2-3"
+
+const int a[] = {1, 2, 3};
+string s = absl::StrJoin(a, "-");
+// s == "1-2-3"
+```
+
+The following example passes a `Formatter` argument to format the pairs in a
+map, using a different separator. This makes the output nice and readable.
+
+{% raw %}
```

Comment:
is the raw block necessary?  the code formatting doesn't kick in?

Reply:
The {% raw %} block is unfortunately necessary because the Jekyll publishing framework interprets "{{" as an instruction :(